numpy/lib/tests/test_function_base.py

```diff
@@ -1499,6 +1500,59 @@ def test_size_zero_output(self):
         with assert_raises_regex(ValueError, 'new output dimensions'):
             f(x)
 
+    @pytest.mark.skipif(not HAS_REFCOUNT, reason="Python lacks refcounts")
+    def test_frompyfunc_leaks(self):
+        # exposed in gh-11867 as np.vectorized, but the problem stems from
+        # frompyfunc.
+        # class.attribute = np.frompyfunc(<method>) creates a
+        # reference cycle that requires a gc collection cycle to break
+        # (on CPython 3)
+        import gc
+        gc.disable()
+        iters = 20
+        try:
+            class A(object):
+                def method(self, *args):
+                    return 0
+
+                @classmethod
+                def method1(*args):
+                    return 0
+
+                def __init__(self):
+                    self.f = np.frompyfunc(self.method, 1, 1)
```

Comment:
Why do this in the constructor, and not as part of the loop? Right now you're creating this even within the unbound method test

Reply:
To elaborate: I think this line belongs after line 1528

Reply:
fixed